### PR TITLE
[#783] unload dialog

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/386611be74686f4/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/67ac82df0a517fe/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/386611be74686f4/output.txt

Total script time: 0.52 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/67ac82df0a517fe/output.txt

Total script time: 0.70 mins

Server is running on http://pr618.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/92c951a6b35ef70/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/2b2000ed0a1c817/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/92c951a6b35ef70/output.txt

Total script time: 0.32 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/2b2000ed0a1c817/output.txt

Total script time: 0.58 mins

Server is running on http://pr618.butyr.org/templates/test.html

File: src/ui/unload-dialog.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    butter.listen( "ready", function() {
+
+      for ( var i = 0, el = events.length; i < el; i++ ) {
+
+        butter.listen( events[ i ], eventFunction );
+      }
+    });
+
+    butter.listen( "projectsaved", function() {
+
```
Comment (secretrobotron) [inline src/ui/unload-dialog.js]: Do you mind fixing up this vertical spacing?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5da56d9efb63113/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/6b54909271065b9/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/5da56d9efb63113/output.txt

Total script time: 0.31 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/6b54909271065b9/output.txt

Total script time: 0.60 mins

Server is running on http://pr618.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/91cc04490e32aea/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/27f1a4703cdc017/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/91cc04490e32aea/output.txt

Total script time: 0.37 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/27f1a4703cdc017/output.txt

Total script time: 0.76 mins

Server is running on http://pr618.butyr.org/templates/test.html

File: src/ui/unload-dialog.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          "trackadded",
+          "trackremoved",
+          "tracktargetchanged",
+          "trackeventadded",
+          "trackeventremoved",
+          "trackeventupdated"
+        ];
+
+    var areYouSure = function() {
+      return "You have unsaved project data.";
```
Comment (secretrobotron) [inline src/ui/unload-dialog.js:18]: I noticed that the message doesn't actually show up in Firefox. This is probably why: https://developer.mozilla.org/en/DOM/window.onbeforeunload#Example

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/783c964fee10933/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/783c964fee10933/output.txt

Total script time: 0.68 mins

Server is running on http://pr618.butyr.org/templates/test.html

**Pull request merged**